script.js

```diff
@@ -0,0 +1,28 @@
+function daysInMonth(month, year) {
+  try {
+    if (!daysAmountInMonth(month, year)) {
```

Comment:
do not call this function (daysAmountInMonth(month, year)) twice.
try to solve without a helper function